base_layer/common_types/src/tari_address/mod.rs

```diff
@@ -152,6 +158,41 @@ impl TariAddress {
         TariAddress::Single(SingleAddress::new_with_interactive_only(spend_key, network))
     }
 
+    pub fn combine_addresses(one: &TariAddress, two: &TariAddress) -> Result<TariAddress, TariAddressError> {
+        if one.comms_public_key() != two.comms_public_key() {
+            return Err(TariAddressError::CreationError("Public keys do not match".to_string()));
+        }
+        if one.network() != two.network() {
+            return Err(TariAddressError::CreationError("Networks do not match".to_string()));
+        }
+        if let TariAddress::Dual(one) = one {
+            if let TariAddress::Dual(two) = two {
+                if one.public_view_key() != two.public_view_key() {
+                    return Err(TariAddressError::CreationError("View keys do not match".to_string()));
+                }
+            }
+        }
+        match (one, two) {
+            (TariAddress::Dual(one), _) => Ok(TariAddress::new_dual_address(
+                one.public_view_key().clone(),
+                one.public_spend_key().clone(),
+                one.network(),
+                one.features().combine_features(two.features()),
+            )),
+            (_, TariAddress::Dual(two)) => Ok(TariAddress::new_dual_address(
+                two.public_view_key().clone(),
+                one.public_spend_key().clone(),
+                one.network(),
```

Comment:
The logic would be more consistent this way:
```suggestion
                two.public_spend_key().clone(),
                two.network(),
```